client/src/search/Typeahead/TypeaheadMenu.js

```diff
@@ -0,0 +1,110 @@
+import { ListGroup, ListGroupItem } from "react-bootstrap";
+
+import { create_text_maker_component } from "../../components";
+
+import text from "./Typeahead.yaml";
+
+const { text_maker, TM } = create_text_maker_component(text);
+
+export class TypeaheadMenu extends React.Component {
+  handleWindowClick = (e) => {
+    const { hide_menu } = this.props;
+    if (!document.getElementsByClassName("rbt")[0].contains(e.target)) {
+      hide_menu();
+    }
+  };
+  componentDidMount() {
+    window.addEventListener("click", this.handleWindowClick);
+  }
+
+  render() {
+    const {
+      search_text,
+      queried_results,
+      config_groups,
+      on_select_item,
+      hide_menu,
+    } = this.props;
+
+    const total_matching_results = queried_results.length;
+
+    let index_key_counter = 0;
+
+    if (_.isEmpty(queried_results)) {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
+          <ListGroupItem className="dropdown-header">
+            {text_maker("no_matches_found")}
+          </ListGroupItem>
+        </ListGroup>
+      );
+    } else {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
```

Comment:
`ListGroup` and `ListGroupItem` are probably not the right choices.

Should probably be a `ul` with `role="listbox"`. Everything inside should be in `li`'s, and the result items should have `role="option"`.